epochStart/metachain/rewardsV2_test.go

```diff
@@ -794,207 +803,231 @@ func TestNewRewardsCreatorV2_computeRewardsPerNode(t *testing.T) {
+	setupResult := setUpRewards(args)
+
+	baseStakePerNode := big.NewInt(0).Mul(big.NewInt(2500), setupResult.smallestDivisionERD)
+	topupStakePerNode := big.NewInt(0).Mul(big.NewInt(750), setupResult.smallestDivisionERD)
+
+	baseEligibleStake := big.NewInt(0).Mul(big.NewInt(1600), baseStakePerNode)
+	topupEligibleStake := big.NewInt(0).Mul(big.NewInt(1600), topupStakePerNode)
+
+	baseAPR := 20.20
+	topupAPR := 13.99
 
 	tests := []struct {
-		rewardsForBlocks *big.Int
-		rewardsForDev    *big.Int
-		baseStake        *big.Int
-		topupStake       *big.Int
-		expectedROI      float64
+		validatorTopupStake      *big.Int
+		validatorExpectedFullROI float64
 	}{
 		{
-			rewardsForBlocks: rewardsForBlocks,
-			rewardsForDev:    big.NewInt(0),
-			baseStake:        big.NewInt(0).Mul(big.NewInt(2500), smallesDivisionERD),
-			topupStake:       big.NewInt(0).Mul(big.NewInt(25), big.NewInt(0).Div(smallesDivisionERD, big.NewInt(100))),
-			expectedROI:      40.67,
+			validatorTopupStake:      big.NewInt(0).Mul(big.NewInt(25), big.NewInt(0).Div(setupResult.smallestDivisionERD, big.NewInt(100))),
+			validatorExpectedFullROI: (baseAPR*2500 + topupAPR*0.25) / 2500.25,
+		},
+		{
+			validatorTopupStake:      big.NewInt(0).Mul(big.NewInt(500), setupResult.smallestDivisionERD),
+			validatorExpectedFullROI: (baseAPR*2500 + topupAPR*500) / 3000,
+		},
+		{
+			validatorTopupStake:      big.NewInt(0).Mul(big.NewInt(1500), setupResult.smallestDivisionERD),
+			validatorExpectedFullROI: (baseAPR*2500 + topupAPR*1500) / 4000,
 		},
 		{
-			rewardsForBlocks: rewardsForBlocks,
-			rewardsForDev:    big.NewInt(0),
-			baseStake:        big.NewInt(0).Mul(big.NewInt(2500), smallesDivisionERD),
-			topupStake:       big.NewInt(0).Mul(big.NewInt(500), smallesDivisionERD),
-			expectedROI:      33.89,
+			validatorTopupStake:      big.NewInt(0).Mul(big.NewInt(2500), setupResult.smallestDivisionERD),
+			validatorExpectedFullROI: (baseAPR*2500 + topupAPR*2500) / 5000,
 		},
 		{
-			rewardsForBlocks: rewardsForBlocks,
-			rewardsForDev:    big.NewInt(0),
-			baseStake:        big.NewInt(0).Mul(big.NewInt(2500), smallesDivisionERD),
-			topupStake:       big.NewInt(0).Mul(big.NewInt(1500), smallesDivisionERD),
-			expectedROI:      25.41,
+			validatorTopupStake:      big.NewInt(0).Mul(big.NewInt(2*2500), setupResult.smallestDivisionERD),
+			validatorExpectedFullROI: (baseAPR*2500 + topupAPR*5000) / 7500,
 		},
 		{
-			rewardsForBlocks: rewardsForBlocks,
-			rewardsForDev:    big.NewInt(0),
-			baseStake:        big.NewInt(0).Mul(big.NewInt(2500), smallesDivisionERD),
-			topupStake:       big.NewInt(0).Mul(big.NewInt(2500), smallesDivisionERD),
-			expectedROI:      20.33,
+			validatorTopupStake:      big.NewInt(0).Mul(big.NewInt(3*2500), setupResult.smallestDivisionERD),
+			validatorExpectedFullROI: (baseAPR*2500 + topupAPR*7500) / 10000,
 		},
 	}
 
-	factor := int64(1000 * 1000)
-	tolerance := 0.01
+	nbEligiblePerShard := uint32(800)
```

Comment:
can we rename to nodesPerShard? because we only have 400 eligible per shard

Reply:
fixed, added also waitingPerShard in order to compute "eligiblePercentage"